### PR TITLE
feat: breakout encryption logic functions

### PR DESCRIPTION
**Pull request by mmcshinsky-bitgo**
* Move encryption functions to seperate files
* Add browser compatible rsa encryption functions

**NOTE:**
The only two new functions in this PR are `encryptBrowserRsaWithAesGcm`, `decryptBrowserRsaWithAesGcm`, and their respective child functions.
* `encrypt-rsa-browser.ts`
* `decrypt-rsa-browser.ts`
* `utils.ts`

PX-3786

TICKET: PX-3786

### CODE REVIEW

File: modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem
Comment (domoneill18) [inline modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem:1]: Why did these keys change?

File: modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts
Comment (domoneill18) [inline modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts:1]: Can we add tests to encrypt via browser version and decrypt via non-browser version? And vice versa (encrypt via non-browser version and decrypt via browser version)?

File: modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem
Comment (mmcshinsky-bitgo) [reply modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem:1]: Was testing against multiple keys.

File: modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts
Comment (mmcshinsky-bitgo) [reply modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts:1]: Got it working manually testing in the browser. Been having a little trouble getting browser testing working correctly in sdk-core

File: modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem
Comment (domoneill18) [reply modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem:1]: So with the original keys it was still working as expected?

File: modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts
Comment (domoneill18) [reply modules/sdk-core/test/unit/bitgo/trading/network/encrypt.ts:1]: Don't you have tests against the browser version in here right now?

File: modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem
Comment (mmcshinsky-bitgo) [reply modules/sdk-core/test/unit/bitgo/trading/network/private-key.pem:1]: Yup! Thought I added them back in just now, but did not 😆

**Review by domoneill18 — DISMISSED** [review]

**Review by zahin-mohammad — DISMISSED** [review]

**Review by domoneill18 — APPROVED** [review]

**Review by margueriteblair — APPROVED** [review]

**Pull request merged**